src/lib/storage/chunk.cpp

```diff
@@ -92,11 +92,11 @@ std::vector<std::shared_ptr<AbstractIndex>> Chunk::get_indexes(
   return result;
 }
 
-void Chunk::finalize() {
+void Chunk::finalize(const bool skip_mvcc_check) {
   Assert(is_mutable(), "Only mutable chunks can be finalized. Chunks cannot be finalized twice.");
   _is_mutable = false;
 
-  if (has_mvcc_data()) {
+  if (has_mvcc_data() && !skip_mvcc_check) {
```

Comment:
@Bensk1 Would it be possible to check if the max_begin_cid was already set and skip the check? That would save use from having this non-obvious argument.

Reply:
That should be possible and I cannot think of a case where it is necessary to re-check for the largest CID and overwrite the `max_begin_cid`. So, yes, ` if (has_mvcc_data() && !_mvcc_data->max_begin_cid) {`, should work.